### PR TITLE
Comment our workspace Try* methods hinting you probably want something else

### PR DESCRIPTION
**Pull request by jasonmalinowski**
The Try* methods are very specialized methods, we should hint that you should call something else.

Review: @agocke, @mattwar, @Pilchie, @DustinCampbell

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/Project.cs [hunk 1]
```diff
@@ -372,7 +372,9 @@ private static TextDocument CreateAdditionalDocument(DocumentId documentId, Proj
         }
 
         /// <summary>
-        /// Get the <see cref="Compilation"/> for this project if it is available.
+        /// Tries to get the cached <see cref="Compilation"/> for this project if it is available. In almost all
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/Workspace/Solution/Project.cs]: Maybe change "if it is available" to "if it has already been computed" (or created)?

**Comment by Pilchie** [comment]
:+1:

**Comment by jaredpar** [comment]
:+1:

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs [hunk 2]
```diff
@@ -200,6 +202,8 @@ public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken =
 
         /// <summary>
         /// Gets the root node of the current syntax tree if it is available.
+        /// In almost all cases, you should call <see cref="GetSyntaxRootAsync"/> to fetch the root node, which will parse
```
Comment (ManishJayaswal) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: If we say in almost all cases we should call the other method then why do we have this method? Should this be changed to "In most cases"?
In general, we have seen several customers hit this issue. Should we add some samples which illustrate the difference?

File: src/Workspaces/Core/Portable/Workspace/Solution/Project.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Project.cs]: Fixed.

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs (hunk 2 above)
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: This method was added for a very special purpose. In hindsight, we either should have named it to something more scary-sounding or just kept it internal. None of our samples will use this -- it's just a matter of people browsing the API in Intellisense, guessing this is what they want, and then trying it.

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs [hunk 3]
```diff
@@ -200,6 +202,8 @@ public Task<SyntaxTree> GetSyntaxTreeAsync(CancellationToken cancellationToken =
 
         /// <summary>
         /// Gets the root node of the current syntax tree if it is available.
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: Is this the only one left that uses the ambiguous "if it is available" wording? Can we update this one too?

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: Is it ambiguous? It seems fine to me...

Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: Yes, to me it is, for the same reason TryGetCompilation was ambiguous (as pointed out by @pilchie earlier). What does it mean to be "available"?

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: To me I'm reading it as meeting the previous criteria of "if the text is already loaded and the tree is already parsed."

But seriously, if it's ambiguous to you, just send a new PR with something that makes more sense to you. I don't say that to be flippant, but I'm not psychic and can't fix problems I don't see.

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: @dpoeschl: what would you like it to say?

**Comment by jasonmalinowski** [comment]
Jenkins, retest this please.

File: src/Workspaces/Core/Portable/Workspace/Solution/Document.cs (hunk 3 above)
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/Document.cs]: Oops, mis-read which line your comment was attached to. Fixed.

**Comment by dpoeschl** [comment]
:+1: (assuming tests pass)

**Comment by jasonmalinowski** [comment]
Jenkins, retest Linux please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot, test Linux please.

**Comment by tannergooding** [comment]
test Linux please

**Pull request merged**